src/agent-tui/forms/regnode.go

```diff
@@ -0,0 +1,91 @@
+package forms
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/gdamore/tcell/v2"
+	tuiNet "github.com/openshift/assisted-installer-agent/src/agent-tui/net"
+	"github.com/openshift/assisted-installer-agent/src/agent-tui/newt"
+	"github.com/rivo/tview"
+)
+
+const (
+	CONNECTIVITYCHECK string = "Check connectivity"
+	NETCONFIGURE      string = "Configure networking"
+	DONE              string = "Done"
+	RENDEZVOUSLABEL   string = "Rendezvous IP Address"
+)
+
+func RegNodeModalForm(app *tview.Application, pages *tview.Pages) tview.Primitive {
+	statusView := tview.NewTextView()
+
+	statusView.SetBackgroundColor(newt.ColorGray).
+		SetBorder(true).
+		SetBorderColor(tcell.ColorBlack).
+		SetTitle("Status").
+		SetTitleColor(tcell.ColorBlack)
+	statusView.SetTextColor(tcell.ColorBlack).
+		SetDynamicColors(true)
+
+	goodConnectivity := false
+	ipField := tview.NewInputField().
+		SetFieldWidth(40).
+		SetLabel(RENDEZVOUSLABEL).
+		SetLabelColor(tcell.ColorBlack)
+
+	ipField.SetDoneFunc(func(key tcell.Key) {
+		if net.ParseIP(ipField.GetText()) == nil {
+			statusView.SetText("[red::b]The specified Rendezvous IP is not a valid IP Address")
+		} else {
+			statusView.Clear()
+		}
+	})
+
+	regNodeConfigForm := tview.NewForm().
+		AddFormItem(ipField).
+		AddButton(CONNECTIVITYCHECK, func() {
+			statusView.Clear()
+			fmt.Fprintf(statusView, "Running connectivity check. Please wait...\n")
+			go func() {
+				addr := ipField.GetText()
+				output, err := tuiNet.ValidateConnectivity(addr)
+				if err != nil {
+					goodConnectivity = false
+					fmt.Fprintf(statusView, "[red::b]Failed to connect to %s[black]\n%s", addr, tview.Escape(string(output)))
```

Comment:
Could also be nice to show the error (if there was some issue running the executable itself)

```suggestion
					fmt.Fprintf(statusView, "[red::b]Failed to connect to %s (%s)[black]\n%s", addr, err, tview.Escape(string(output)))
```

Reply:
Agreed